generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/SchemaGenerator.kt

```diff
@@ -70,24 +70,21 @@ open class SchemaGenerator(internal val config: SchemaGeneratorConfig) : Closeab
             codeRegistry.fieldVisibility(NoIntrospectionGraphqlFieldVisibility.NO_INTROSPECTION_FIELD_VISIBILITY)
         }
 
-        return GraphQLSchema.newSchema()
-            .query(generateQueries(this, queries))
-            .mutation(generateMutations(this, mutations))
-            .subscription(generateSubscriptions(this, subscriptions))
-            .additionalTypes(generateAdditionalTypes())
-            .additionalDirectives(directives.values.toSet())
-            .codeRegistry(codeRegistry.build())
+        return config.hooks.willBuildSchema(queries, mutations, subscriptions, additionalTypes, additionalInputTypes, schemaObject)
             .also { builder ->
```

Comment:
instead of applying this logic at the end which required clearing up the directives and manually re-adding generated stuff -> we could just move it up front